Remove SHA224 from webkit.

nmahendru opened the pull request:
#### 974df4cfe6979ab97bab8e7d93f4e1389de5716d
<pre>
Remove SHA224 from webkit.
<a href="https://bugs.webkit.org/show_bug.cgi?id=278197">https://bugs.webkit.org/show_bug.cgi?id=278197</a>
<a href="https://rdar.apple.com/134533233">rdar://134533233</a>

Reviewed by David Kilzer.

Removing all references to sha224 in WebKit.
There are no uses of it in the current code and it&apos;s been removed
from the WebCrypto standard since 2017.
Firefox/Chrome don&apos;t support it and CryptoKit also does not support it,
forcing us to have the internal interface return an optional for CryptoKit.
If, down the line, we find some use case for it then we can revert this.
It&apos;s highly unlikely though.

* LayoutTests/TestExpectations:
* LayoutTests/crypto/subtle/rsa-pss-import-key-sign-large-salt-expected.txt:
* LayoutTests/crypto/subtle/rsa-pss-import-key-sign-large-salt.html:
* LayoutTests/crypto/subtle/hmac-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-generate-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-generate-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-import-jwk-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-import-jwk-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/hmac-import-key-sign-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-sign-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-verify-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-verify-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/rsa-oaep-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-oaep-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-oaep-import-jwk-public-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-oaep-import-jwk-public-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-pss-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-pss-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-pss-import-jwk-public-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-pss-import-jwk-public-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-jwk-public-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-jwk-public-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-sign-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-sign-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-verify-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-verify-sha224.html: Removed.
* LayoutTests/crypto/subtle/sha224-digest-expected.txt: Removed.
* LayoutTests/crypto/subtle/sha224-digest.html: Removed.
* LayoutTests/crypto/workers/subtle/resources/sha224-digest.js: Removed.
* LayoutTests/crypto/workers/subtle/sha224-digest-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/sha224-digest.html: Removed.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/win/TestExpectations:
* Source/WTF/wtf/spi/cocoa/SecuritySPI.h:
* Source/WebCore/PAL/pal/crypto/CryptoDigest.h:
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
(PAL::toSHA1Context):
(PAL::createCryptoDigest):
(PAL::CryptoDigest::addBytes):
(PAL::CryptoDigest::computeHash):
(PAL::toSHA224Context): Deleted.
* Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
(PAL::getGcryptAlgorithm):
* Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
(PAL::createCryptoDigest):
* Source/WebCore/PAL/pal/spi/cocoa/CommonCryptoSPI.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/CommonCryptoUtilities.cpp:
(WebCore::getCommonCryptoDigestAlgorithm):
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHMAC.cpp:
(WebCore::CryptoAlgorithmHMAC::importKey):
(WebCore::CryptoAlgorithmHMAC::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.cpp:
(WebCore::CryptoAlgorithmRSASSA_PKCS1_v1_5::importKey):
(WebCore::CryptoAlgorithmRSASSA_PKCS1_v1_5::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_OAEP.cpp:
(WebCore::CryptoAlgorithmRSA_OAEP::importKey):
(WebCore::CryptoAlgorithmRSA_OAEP::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_PSS.cpp:
(WebCore::CryptoAlgorithmRSA_PSS::importKey):
(WebCore::CryptoAlgorithmRSA_PSS::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.cpp: Removed.
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.h: Removed.
* Source/WebCore/crypto/cocoa/CryptoAlgorithmHKDFMac.cpp:
(WebCore::CryptoAlgorithmHKDF::platformDeriveBits):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmHMACMac.cpp:
(WebCore::platformSignCryptoKit):
(WebCore::platformVerifyCryptoKit):
(WebCore::commonCryptoHMACAlgorithm):
(WebCore::CryptoAlgorithmHMAC::platformSign):
(WebCore::CryptoAlgorithmHMAC::platformVerify):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmPBKDF2Mac.cpp:
(WebCore::CryptoAlgorithmPBKDF2MacInternal::commonCryptoHMACAlgorithm):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/cocoa/CryptoDigestAlgorithm.h:
(WebCore::cryptoDigestAlgorithm):
* Source/WebCore/crypto/cocoa/CryptoUtilitiesCocoa.cpp:
(WebCore::calculateHMACSignature):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmHMACGCrypt.cpp:
(WebCore::getGCryptDigestAlgorithm):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/gcrypt/GCryptUtilities.cpp:
(WebCore::hashAlgorithmName):
(WebCore::hmacAlgorithm):
(WebCore::digestAlgorithm):
(WebCore::hashCryptoDigestAlgorithm):
* Source/WebCore/crypto/keys/CryptoKeyHMAC.cpp:
(WebCore::getKeyLengthFromHash):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/OpenSSLUtilities.cpp:
(WebCore::digestAlgorithm):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Tools/TestWebKitAPI/Tests/WebCore/CryptoDigest.cpp:
(TestWebKitAPI::expectSHA224): Deleted.
(TestWebKitAPI::TEST(CryptoDigest, SHA224Computation)): Deleted.

Canonical link: <a href="https://commits.webkit.org/290341@main">https://commits.webkit.org/290341@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1e5e16c4a5cf201fcf573f852bd3077eee022211

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89582 "3 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/9109 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44442 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94574 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40349 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/91634 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9498 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17388 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/69003 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26656 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92583 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7289 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81299 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49367 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/7020 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35684 "Found 60 new test failures: compositing/animation/repaint-after-clearing-shared-backing.html fast/repaint/switch-overflow-vertical-rl.html fonts/monospace.html fonts/sans-serif.html fonts/serif.html fonts/unicode-character-font-crash.html fonts/use-typo-metrics-1.html fullscreen/exit-full-screen-video-crash.html fullscreen/full-screen-document-background-color.html http/tests/security/contentSecurityPolicy/block-all-mixed-content/insecure-script-in-iframe.html ... (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39455 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/82380 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77374 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36688 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96402 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/88357 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16764 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12317 "Found 1 new test failure: imported/w3c/web-platform-tests/html/browsers/the-window-object/open-close/open-features-tokenization-noreferrer.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77879 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/17019 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/77102 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77193 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21624 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20204 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9931 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14081 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16777 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/22091 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/110850 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16518 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26569 "Found 20 new JSC stress test failures: microbenchmarks/memcpy-wasm-medium.js.default, microbenchmarks/memcpy-wasm-medium.js.dfg-eager, microbenchmarks/memcpy-wasm-medium.js.mini-mode, microbenchmarks/memcpy-wasm-small.js.dfg-eager, microbenchmarks/memcpy-wasm-small.js.mini-mode, microbenchmarks/set-delete-add.js.dfg-eager-no-cjit-validate, stress/sampling-profiler-display-name.js.dfg-eager-no-cjit-validate, wasm.yaml/wasm/function-tests/memcpy-wasm-loop.js.default-wasm, wasm.yaml/wasm/function-tests/memcpy-wasm-loop.js.wasm-collect-continuously, wasm.yaml/wasm/stress/ipint-bbq-osr-with-try5.js.wasm-collect-continuously ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19969 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18300 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
```diff
@@ -129,9 +125,10 @@ static std::variant<std::unique_ptr<CC_SHA1_CTX>, std::unique_ptr<CC_SHA256_CTX>
         std::unique_ptr<CC_SHA256_CTX> context = WTF::makeUniqueWithoutFastMallocCheck<CC_SHA256_CTX>();
-        CC_SHA224_Init(context.get());
+        CC_SHA256_Init(context.get());
         return context;
```
None of the code after the `RELEASE_ASSERT()` is reachable.  We should just delete it, or does the compiler complain?

nmahendru replied:
I have kept it for the compiler.

ddkilzer commented on Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
```diff
@@ -172,8 +169,9 @@ void CryptoDigest::addBytes(std::span<const uint8_t> input)
+    case CryptoDigest::Algorithm::DEPRECATED_SHA_224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE("SHA224 is not supported.");
+        CC_SHA256_Update(toSHA256Context(m_context.get()), static_cast<const void*>(input.data()), input.size());
         return;
```
Ditto regarding unreachable code.

ddkilzer commented on Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
```diff
@@ -50,8 +50,9 @@ static int getGcryptAlgorithm(CryptoDigest::Algorithm algorithm)
         return GCRY_MD_SHA1;
-    case CryptoDigest::Algorithm::SHA_224:
-        return GCRY_MD_SHA224;
+    case CryptoDigest::Algorithm::DEPRECATED_SHA_224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE("SHA224 is not supported.");
+        return GCRY_MD_SHA256;
```
Ditto regarding unreachable code.

ddkilzer commented on Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
```diff
@@ -112,8 +104,9 @@ static std::unique_ptr<CryptoDigestContext> createCryptoDigest(CryptoDigest::Alg
-        return CryptoDigestContextImpl<SHA256_CTX, SHA224Functions>::create();
+    case CryptoDigest::Algorithm::DEPRECATED_SHA_224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE("SHA224 is not supported.");
+        return CryptoDigestContextImpl<SHA256_CTX, SHA256Functions>::create();
```
Ditto regarding unreachable code.

ddkilzer commented on Source/WebCore/crypto/CommonCryptoUtilities.cpp:
```diff
@@ -36,8 +36,9 @@ ALLOW_DEPRECATED_DECLARATIONS_BEGIN
         return true;
-    case CryptoAlgorithmIdentifier::SHA_224:
-        algorithm = kCCDigestSHA224;
+    case CryptoAlgorithmIdentifier::DEPRECATED_SHA_224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE(sha224DeprecationMessage);
+        algorithm = kCCDigestSHA256;
         return true;
```
Ditto regarding unreachable code.

ddkilzer commented on Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
```diff
@@ -27,6 +27,7 @@
 
 namespace WebCore {
 
+constexpr auto sha224DeprecationMessage = "SHA224 is not supported";
```
Should be using the `_s` operator here instead?  I guess if this is `constexpr`, it's probably equivalent.

```suggestion
auto sha224DeprecationMessage = "SHA224 is not supported"_s;
```

nmahendru replied:
that fails to compile with "....must be initialized by a constant expression"

ddkilzer commented on Source/WebCore/crypto/cocoa/CryptoAlgorithmHMACMac.cpp:
```diff
@@ -36,13 +36,30 @@
-        return kCCHmacAlgSHA224;
+    case CryptoAlgorithmIdentifier::DEPRECATED_SHA_224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE(sha224DeprecationMessage);
+        return kCCHmacAlgSHA1;
```
Don't we "upgrade" to return SHA256 above instead of SHA1?

nmahendru replied:
I have just used sha256 as the return value to keep compiler happy. I will change the SHA1 to sha256 just to keep that convention true. Although, it does not matter.

ddkilzer commented on Source/WebCore/crypto/cocoa/CryptoAlgorithmPBKDF2Mac.cpp:
```diff
@@ -38,8 +38,9 @@ static CCPseudoRandomAlgorithm commonCryptoHMACAlgorithm(CryptoAlgorithmIdentifi
-        return kCCPRFHmacAlgSHA224;
+    case CryptoAlgorithmIdentifier::DEPRECATED_SHA_224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE(sha224DeprecationMessage);
+        return kCCPRFHmacAlgSHA1;
```
Ditto SHA224 -> SHA256

ddkilzer commented on Source/WebCore/crypto/cocoa/CryptoUtilitiesCocoa.cpp:
```diff
@@ -146,6 +146,7 @@ Vector<uint8_t> calculateHMACSignature(CCHmacAlgorithm algorithm, const Vector<u
         break;
     case kCCHmacAlgSHA224:
+        RELEASE_ASSERT_NOT_REACHED_WITH_MESSAGE(sha224DeprecationMessage);
         digestLength = CC_SHA224_DIGEST_LENGTH;
```
Should `CC_SHA224_DIGEST_LENGTH` be renamed to `DEPRECATED_CC_SHA224_DIGEST_LENGTH`?

nmahendru replied:
That is a commoncrypto enum which i can't change. I will use 256 here to just follow that convention.

ddkilzer reviewed (approved):
r=me, but please consider addressing feedback comments if they make sense

nmahendru committed:
Remove SHA224 from webkit.
https://bugs.webkit.org/show_bug.cgi?id=278197
rdar://134533233

Reviewed by David Kilzer.

Removing all references to sha224 in WebKit.
There are no uses of it in the current code and it's been removed
from the WebCrypto standard since 2017.
Firefox/Chrome don't support it and CryptoKit also does not support it,
forcing us to have the internal interface return an optional for CryptoKit.
If, down the line, we find some use case for it then we can revert this.
It's highly unlikely though.

* LayoutTests/TestExpectations:
* LayoutTests/crypto/subtle/rsa-pss-import-key-sign-large-salt-expected.txt:
* LayoutTests/crypto/subtle/rsa-pss-import-key-sign-large-salt.html:
* LayoutTests/crypto/subtle/hmac-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-generate-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-generate-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-import-jwk-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-import-jwk-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/hmac-import-key-sign-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-sign-sha224.html: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-verify-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/hmac-import-key-verify-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-import-key-malformed-parameters-expected.txt:
* LayoutTests/crypto/subtle/rsa-oaep-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-oaep-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-oaep-import-jwk-public-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-oaep-import-jwk-public-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-pss-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-pss-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsa-pss-import-jwk-public-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsa-pss-import-jwk-public-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-generate-export-key-jwk-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-generate-export-key-jwk-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-jwk-public-key-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-jwk-public-key-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-sign-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-sign-sha224.html: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-verify-sha224-expected.txt: Removed.
* LayoutTests/crypto/subtle/rsassa-pkcs1-v1_5-import-key-verify-sha224.html: Removed.
* LayoutTests/crypto/subtle/sha224-digest-expected.txt: Removed.
* LayoutTests/crypto/subtle/sha224-digest.html: Removed.
* LayoutTests/crypto/workers/subtle/resources/sha224-digest.js: Removed.
* LayoutTests/crypto/workers/subtle/sha224-digest-expected.txt: Removed.
* LayoutTests/crypto/workers/subtle/sha224-digest.html: Removed.
* LayoutTests/platform/glib/TestExpectations:
* LayoutTests/platform/ios/TestExpectations:
* LayoutTests/platform/mac/TestExpectations:
* LayoutTests/platform/win/TestExpectations:
* Source/WTF/wtf/spi/cocoa/SecuritySPI.h:
* Source/WebCore/PAL/pal/crypto/CryptoDigest.h:
* Source/WebCore/PAL/pal/crypto/commoncrypto/CryptoDigestCommonCrypto.cpp:
(PAL::toSHA1Context):
(PAL::createCryptoDigest):
(PAL::CryptoDigest::addBytes):
(PAL::CryptoDigest::computeHash):
(PAL::toSHA224Context): Deleted.
* Source/WebCore/PAL/pal/crypto/gcrypt/CryptoDigestGCrypt.cpp:
(PAL::getGcryptAlgorithm):
* Source/WebCore/PAL/pal/crypto/openssl/CryptoDigestOpenSSL.cpp:
(PAL::createCryptoDigest):
* Source/WebCore/PAL/pal/spi/cocoa/CommonCryptoSPI.h:
* Source/WebCore/Sources.txt:
* Source/WebCore/WebCore.xcodeproj/project.pbxproj:
* Source/WebCore/bindings/js/SerializedScriptValue.cpp:
(WebCore::CloneSerializer::write):
(WebCore::CloneDeserializer::read):
* Source/WebCore/crypto/CommonCryptoUtilities.cpp:
(WebCore::getCommonCryptoDigestAlgorithm):
* Source/WebCore/crypto/CryptoAlgorithmIdentifier.h:
* Source/WebCore/crypto/SubtleCrypto.cpp:
(WebCore::normalizeCryptoAlgorithmParameters):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmHMAC.cpp:
(WebCore::CryptoAlgorithmHMAC::importKey):
(WebCore::CryptoAlgorithmHMAC::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSASSA_PKCS1_v1_5.cpp:
(WebCore::CryptoAlgorithmRSASSA_PKCS1_v1_5::importKey):
(WebCore::CryptoAlgorithmRSASSA_PKCS1_v1_5::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_OAEP.cpp:
(WebCore::CryptoAlgorithmRSA_OAEP::importKey):
(WebCore::CryptoAlgorithmRSA_OAEP::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmRSA_PSS.cpp:
(WebCore::CryptoAlgorithmRSA_PSS::importKey):
(WebCore::CryptoAlgorithmRSA_PSS::exportKey):
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.cpp: Removed.
* Source/WebCore/crypto/algorithms/CryptoAlgorithmSHA224.h: Removed.
* Source/WebCore/crypto/cocoa/CryptoAlgorithmHKDFMac.cpp:
(WebCore::CryptoAlgorithmHKDF::platformDeriveBits):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmHMACMac.cpp:
(WebCore::platformSignCryptoKit):
(WebCore::platformVerifyCryptoKit):
(WebCore::commonCryptoHMACAlgorithm):
(WebCore::CryptoAlgorithmHMAC::platformSign):
(WebCore::CryptoAlgorithmHMAC::platformVerify):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmPBKDF2Mac.cpp:
(WebCore::CryptoAlgorithmPBKDF2MacInternal::commonCryptoHMACAlgorithm):
* Source/WebCore/crypto/cocoa/CryptoAlgorithmRegistryMac.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/cocoa/CryptoDigestAlgorithm.h:
(WebCore::cryptoDigestAlgorithm):
* Source/WebCore/crypto/cocoa/CryptoUtilitiesCocoa.cpp:
(WebCore::calculateHMACSignature):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmHMACGCrypt.cpp:
(WebCore::getGCryptDigestAlgorithm):
* Source/WebCore/crypto/gcrypt/CryptoAlgorithmRegistryGCrypt.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/gcrypt/GCryptUtilities.cpp:
(WebCore::hashAlgorithmName):
(WebCore::hmacAlgorithm):
(WebCore::digestAlgorithm):
(WebCore::hashCryptoDigestAlgorithm):
* Source/WebCore/crypto/keys/CryptoKeyHMAC.cpp:
(WebCore::getKeyLengthFromHash):
* Source/WebCore/crypto/openssl/CryptoAlgorithmRegistryOpenSSL.cpp:
(WebCore::CryptoAlgorithmRegistry::platformRegisterAlgorithms):
* Source/WebCore/crypto/openssl/OpenSSLUtilities.cpp:
(WebCore::digestAlgorithm):
* Source/WebKit/Shared/WebCoreArgumentCoders.serialization.in:
* Tools/TestWebKitAPI/Tests/WebCore/CryptoDigest.cpp:
(TestWebKitAPI::expectSHA224): Deleted.
(TestWebKitAPI::TEST(CryptoDigest, SHA224Computation)): Deleted.

Canonical link: https://commits.webkit.org/290341@main
